scala/scala_import.bzl

```diff
@@ -93,29 +93,28 @@ def _code_jars_and_intellij_metadata_from(jars, srcjar):
     intellij_metadata = []
     for jar in jars:
         current_jar_code_jars = _filter_out_non_code_jars(jar.files)
-        (current_jar_source_jars, src_jar) = _source_jars(jar, srcjar)
+        current_jar_source_jars = _source_jars(jar, srcjar)
         code_jars += current_jar_code_jars
         for current_class_jar in current_jar_code_jars:  #intellij, untested
             intellij_metadata.append(
                 struct(
                     ijar = None,
                     class_jar = current_class_jar,
-                    source_jar = src_jar,
                     source_jars = current_jar_source_jars,
                 ),
             )
     return struct(code_jars = code_jars, intellij_metadata = intellij_metadata)
 
 def _source_jars(jar, srcjar):
     if srcjar:
-        return ([], srcjar)
+        return [srcjar]
     else:
         jar_source_jars = [
             file
             for file in jar.files.to_list()
             if _is_source_jar(file)
         ]
-        return (jar_source_jars, None)
+        return jar_source_jars
```

Comment:
Why don’t we merge srcjar into the result list? I think this is why thrift sources at Stripe don’t show up with sources: they use code generation which works via source jars.